packages/@react-aria/interactions/test/useFocus.test.js

```diff
@@ -153,4 +154,157 @@ describe('useFocus', function () {
     // MutationObserver is async
     await waitFor(() => expect(onBlur).toHaveBeenCalled());
   });
+
+  describe('useFocus with Shadow DOM', function () {
+    it('handles focus events', function () {
+      const {shadowRoot, shadowHost} = createShadowRoot();
+      const events = [];
+      const ExampleComponent = () => (
+        <Example
+          onFocus={(e) => events.push({type: 'focus', target: e.target})}
+          onBlur={(e) => events.push({type: 'blur', target: e.target})}
+          onFocusChange={isFocused => events.push({type: 'focuschange', isFocused})} />
+      );
+
+      let root;
+      act(() => {
+        root = reactDomRenderer(<ExampleComponent />, shadowRoot);
+      });
```

Comment:
I think you could restructure these and drop the reactDomRenderer if you did something along these lines

```
      const {shadowRoot, shadowHost} = createShadowRoot();
      const ExampleComponent = () => ReactDOM.createPortal(
        <Example
          onFocus={(e) => events.push({type: 'focus', target: e.target})}
          onBlur={(e) => events.push({type: 'blur', target: e.target})}
          onFocusChange={isFocused => events.push({type: 'focuschange', isFocused})} />,
         shadowRoot
      );
      let root = render(<ExampleComponent />);
```

Might also allow you to get rid of the extra act and it gives you a cleanup function on the react tree. It might also solve the setTimeout(..., 0) you have in here too

Unless the distinction of having the react root being inside the shadow dom matters?

Reply:
I updated the tests that used `reactDomRenderer` and removed it.